src/osx/cocoa/desktopenv.mm

```diff
@@ -0,0 +1,43 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/osx/cocoa/desktopenv.cpp
+// Purpose:     wxDesktopEnv
+// Author:      Igor Korot
+// Modified by:
+// Created:     17/11/15
+// Copyright:   (c) wxWidgets team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+#import <Foundation/NSArray.h>
+#import <Foundation/NSURL.h>
+#import <Foundation/NSFileManager.h>
+#import <AppKit/NSWorkspace.h>
+
+#include "wx/filefn.h"
+#include "wx/log.h"
+#include "wx/osx/core/cfstring.h"
+#include "wx/desktopenv.h"
+
+#include "wx/osx/private.h"
+
+wxDesktopEnv::wxDesktopEnv()
+{
+}
+
+wxDesktopEnv::~wxDesktopEnv()
+{
+}
+
+bool wxDesktopEnv::MoveFileToRecycleBin(const wxString &fileName)
+{
+    bool ret = true;
```

Comment:
I would initialize to false, and only set to true when moving to trash is successful.